internal/actions/sync_branch.go

```diff
@@ -347,21 +268,59 @@ func syncBranchRebase(
 		return &SyncBranchResult{
 			*rebase,
 			&SyncBranchContinuation{
-				OldHead:      branchHead,
-				ParentCommit: parentHead,
+				OldHead: branchHead,
 			},
 			branch,
+			false,
 		}, nil
-	case git.RebaseUpdated:
-		return &SyncBranchResult{*rebase, nil, branch}, nil
 	default:
-		// We shouldn't even get an already-up-to-date or not-in-progress
-		// here...
-		logrus.Warn("unexpected rebase status: ", rebase.Status)
-		return &SyncBranchResult{*rebase, nil, branch}, nil
+		return &SyncBranchResult{
+			*rebase,
+			nil,
+			branch,
+			false,
+		}, nil
 	}
 }
 
+func hasLeftover(repo *git.Repo, branch meta.Branch, opts SyncBranchOpts) (bool, error) {
+	// The PR is merged. We get the rebase upstream (more strictly speaking the --onto
+	// part) and check if currently the branch points to the same commit. If they are
+	// pointing to the same one, no leftover.
+	var rebaseUpstream string
+	if branch.IsStackRoot() {
+		if opts.ToTrunk {
+			trunk := branch.Parent.Name
+			trunkHead, err := repo.RevParse(&git.RevParse{Rev: "origin/" + trunk})
+			if err != nil {
+				return false, errors.WrapIff(err, "failed to get HEAD of %q", trunk)
+			}
+			rebaseUpstream = trunkHead
+		} else if branch.MergeCommit == "" {
+			// The PR is closed without merge. There are two possibilities (1) the PR is
```

Comment:
This comment is incorrect. We set `branch.MergeCommit` correctly when merged with FF mode.

See https://github.com/aviator-co/av/blob/f2240b0486022b968fcc0823edb1b99e2825cc2b/internal/actions/pr.go#L508 and https://github.com/aviator-co/av/blob/f2240b0486022b968fcc0823edb1b99e2825cc2b/internal/gh/pullrequest.go#L56-L65.

Reply:
This also hinges on functionality in MergeQueue where we create the FF merge commit with the text "Closes #<number>" which generates the requisite timeline entry.